api/bazel/api_build_system.bzl

```diff
@@ -180,3 +181,60 @@ def api_go_test(name, size, importpath, srcs = [], deps = []):
         importpath = importpath,
         deps = deps,
     )
+
+_GO_BAZEL_RULE_MAPPING = {
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_proto_go",
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_config_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_and_config_proto_go",
+    "@com_google_googleapis//google/api/expr/v1alpha1:syntax_proto": "@com_google_googleapis//google/api/expr/v1alpha1:cel_go_proto",
+}
+
+def go_proto_mapping(dep):
+    mapped = _GO_BAZEL_RULE_MAPPING.get(dep)
+    if mapped == None:
+        return _Suffix("@" + Label(dep).workspace_name + "//" + Label(dep).package + ":" + Label(dep).name, _GO_PROTO_SUFFIX)
+    return mapped
+
+def api_proto_package(name, srcs = [], deps = [], has_services = False, visibility = ["//visibility:public"]):
+    native.proto_library(
+        name = name,
+        srcs = srcs,
+        deps = deps + [
+            "@com_google_protobuf//:any_proto",
+            "@com_google_protobuf//:descriptor_proto",
+            "@com_google_protobuf//:duration_proto",
+            "@com_google_protobuf//:empty_proto",
+            "@com_google_protobuf//:struct_proto",
+            "@com_google_protobuf//:timestamp_proto",
+            "@com_google_protobuf//:wrappers_proto",
+            "@com_google_googleapis//google/api:http_proto",
+            "@com_google_googleapis//google/api:annotations_proto",
+            "@com_google_googleapis//google/rpc:status_proto",
+            "@com_github_gogo_protobuf//:gogo_proto",
+            "@com_envoyproxy_protoc_gen_validate//validate:validate_proto",
+        ],
+        visibility = visibility,
+    )
+
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "//bazel:pgv_plugin_go"]
+    if has_services:
+        compilers = ["@io_bazel_rules_go//proto:go_grpc", "//bazel:pgv_plugin_go"]
+
+    go_proto_library(
+        name = _Suffix(name, _GO_PROTO_SUFFIX),
+        compilers = compilers,
+        importpath = _Suffix(_GO_IMPORTPATH_PREFIX, native.package_name()),
+        proto = name,
+        visibility = ["//visibility:public"],
+        deps = [go_proto_mapping(dep) for dep in deps] + [
+            "@com_github_gogo_protobuf//:gogo_proto_go",
+            "@com_github_golang_protobuf//ptypes:go_default_library",
+            "@com_github_golang_protobuf//ptypes/any:go_default_library",
```

Comment:
Is it possible to have a single WKT list (or least fewer than the current number) in this file and generate the language specific deps (including C++) from this?

Reply:
I'm not sure I follow. We could do the same thing for other languages by mapping WKT to cc/py_library, but not until we migrate the clients here and elsewhere. Due to ODR, you cannot have two definitions for the protos.

Reply:
I guess I'm saying that all the places in this file that talk about WKT want "duration, struct, timestamp, ...". Should this list be in one place and then we do the mangling to the Go, proto, C++ specific dependencies progamatically?

Reply:
I pulled out protos themselves, but language mapping is pretty quirky. Go, for example, adds dependencies that are not coming from protos. It also combines multiple proto libraries into a single go library. Since it's not uniform, I think we should keep specifics to the language.